juju/model.py

```diff
@@ -495,6 +495,11 @@ def __init__(self, charmstore, get_series):
         self.charmstore = charmstore
         self.get_series = get_series
 
+    @staticmethod
+    def _default_app_name(meta):
+        suggested_name = meta.get('charm-metadata', {}).get('Name')
+        return suggested_name or meta.get('id', {}).get('Name')
+
```

Comment:
charmhub api will never again return `charm-metadata`.  maybe its silly to even support this.  until the day someone say "turn it back on".